languages/python/oso/oso/oso.py

```diff
@@ -80,6 +167,66 @@ def get_allowed_actions(self, actor, resource, allow_wildcard=False) -> list:
 
         return list(actions)
 
+    def authorize_field(self, actor, action, resource, field):
+        """Ensure that ``actor`` is allowed to perform ``action`` on a given
+        ``resource``'s ``field``.
+
+        If the action is permitted by an ``allow_field`` rule in the policy,
+        then this method returns ``None``. If the action is not permitted by the
+        policy, this method will raise a ``ForbiddenError``.
+
+        :param actor: The actor performing the request.
+        :param action: The action the actor is attempting to perform on the
+            field.
+        :param resource: The resource being accessed.
+        :param field: The name of the field being accessed.
+        """
+        if not self.query_rule_once("allow_field", actor, action, resource, field):
+            raise self.forbidden_error()
+
+    def authorized_fields(
+        self, actor, action, resource, allow_wildcard=False
+    ) -> List[Any]:
+        """Determine the fields of ``resource`` on which ``actor`` is allowed to
+        perform  ``action``.
+
+        Uses ``allow_field`` rules in the policy to find all allowed fields.
+
+        :param actor: The actor for whom to collect allowed fields.
+        :param action: The action being taken on the field.
+        :param resource: The resource being accessed.
+
+        :param allow_wildcard: Flag to determine behavior if the policy \
+            includes a wildcard field. E.g., a rule allowing any field: \
+            ``allow_field(_actor, _action, _resource, _field)``. If ``True``, the \
+            method will return ``["*"]``, if ``False``, the method will raise an \
+            exception.
+
+        :type allow_wildcard: bool
+
+        :return: A list of the unique allowed fields.
+        """
+        results = self.query_rule(
+            "allow_field", actor, action, resource, Variable("field")
+        )
+        fields = set()
+        for result in results:
+            field = result.get("bindings").get("field")
+            if isinstance(field, Variable):
+                if not allow_wildcard:
+                    raise exceptions.OsoError(
+                        """The result of authorized_fields() contained an
+                        "unconstrained" field that could represent any
+                        field, but allow_wildcard was set to False. To fix,
+                        set allow_wildcard to True and compare with the "*"
+                        string."""
+                    )
+                else:
+                    return ["*"]
+            fields.add(field)
+
+        return list(fields)
```

Comment:
Same comment as https://github.com/osohq/oso/pull/1102#discussion_r696953414

Reply:
Hmm this will make for a breaking change -- perhaps I'll have `get_allowed_actions` return `list(self.authorized_actions(...))`

Reply:
Up to you — we're making a bunch of other breaking changes in 0.20.0 so it wouldn't be terrible to add one more (small) one onto the pile.